### PR TITLE
Fixed cognitive complexity for SARIF code

### PR DESCRIPTION
**Pull request by sadmananik**

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java [hunk 1]
```diff
@@ -303,78 +303,87 @@ public void removeTemporaryInputJson(List<String> commandLineArgs) {
 
     public Sarif prepareSarifObject(Set<String> securityProducts, Map<String, Object> scanParameters) {
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java:304]: Unit test is missing for this method

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java [hunk 2]
```diff
@@ -303,78 +303,87 @@ public void removeTemporaryInputJson(List<String> commandLineArgs) {
...
-                sarif.getIssue().setTypes(issueTypes);
-            }
+            handlePolaris(scanParameters, sarif);
             return sarif;
         }
-
         return null;
     }
+
+    private void handleBlackDuck(Map<String, Object> scanParameters, Sarif sarif) {
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java:318]: We can return **Sarif** object from this method and making it public, unit test can be added to increase test coverage

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java [hunk 3]
```diff
@@ -303,78 +303,87 @@ public void removeTemporaryInputJson(List<String> commandLineArgs) {
...
+            sarif.setSeverities(severities);
+        }
+        if (scanParameters.containsKey(ApplicationConstants.BLACKDUCK_REPORTS_SARIF_GROUPSCAISSUES_KEY)) {
+            Boolean reports_sarif_groupSCAIssues =
+                    (Boolean) scanParameters.get(ApplicationConstants.BLACKDUCK_REPORTS_SARIF_GROUPSCAISSUES_KEY);
+            sarif.setGroupSCAIssues(reports_sarif_groupSCAIssues);
+        }
+    }
+
+    private void handlePolaris(Map<String, Object> scanParameters, Sarif sarif) {
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/service/ScannerArgumentService.java:346]: We can return **Sarif** object from this method and making it public, unit test can be added to increase test coverage

File: src/main/java/io/jenkins/plugins/synopsys/security/scan/global/Utility.java [hunk 4]
```diff
@@ -216,4 +217,34 @@ public static String jenkinsJobType(EnvVars envVars) {
             return "UnknownJobType";
         }
     }
+
+    public static String getDefaultSarifReportFilePath(boolean isBlackDuckScan, boolean isPolarisDuckScan) {
```
Comment (zaman-akib) [inline src/main/java/io/jenkins/plugins/synopsys/security/scan/global/Utility.java:221]: Add unit tests for all the new methods (there are 4 including this) of this class

**Review by zaman-akib — CHANGES_REQUESTED** [review]

**Review by zaman-akib — APPROVED** [review]

**Review by jahid1209 — APPROVED** [review]

**Pull request merged**